src/main/java/com/minecolonies/entity/EntityCitizen.java

```diff
@@ -1263,6 +1315,49 @@ private void hitBlockWithToolInHand(@Nullable final BlockPos blockPos, final boo
         }
     }
 
+    @Override
+    public boolean attackEntityFrom(@NotNull DamageSource damageSource, float damage)
+    {
+        Entity sourceEntity = damageSource.getEntity();
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Local variable 'sourceEntity' could be declared final [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=pmd%3ALocalVariableCouldBeFinal)